### PR TITLE
Start moving all wobbly line logic into the React component

### PR DESCRIPTION
**Pull request by davidpmccormick**
Getting the wobbly line to look after itself in React. WIP.

### CODE REVIEW

**Comment by jamesgorrie** [comment]
Loads of conflicts.

File: catalogue/webapp/yarn-error.log [hunk 1]
```diff
@@ -5723,13 +5723,13 @@ Trace:
...
   
-      at ProcessTermError.MessageError (/Users/gorriej/.yarn/lib/cli.js:186:110)
-      at new ProcessTermError (/Users/gorriej/.yarn/lib/cli.js:226:113)
-      at ChildProcess.<anonymous> (/Users/gorriej/.yarn/lib/cli.js:30281:17)
+      at ProcessTermError.MessageError (/usr/local/Cellar/yarn/1.5.1_1/libexec/lib/cli.js:186:110)
+      at new ProcessTermError (/usr/local/Cellar/yarn/1.5.1_1/libexec/lib/cli.js:226:113)
+      at ChildProcess.<anonymous> (/usr/local/Cellar/yarn/1.5.1_1/libexec/lib/cli.js:30281:17)
       at emitTwo (events.js:126:13)
       at ChildProcess.emit (events.js:214:7)
       at maybeClose (internal/child_process.js:925:16)
```
Comment (jamesgorrie) [inline catalogue/webapp/yarn-error.log]: Let's get rid of this.

File: common/views/components/WobblyEdge/WobblyEdge.js [hunk 2]
```diff
@@ -1,18 +1,93 @@
...
+    this.state = {
+      isActive: false,
+      styleObject: prefixedPropertyStyleObject('clipPath', this.makePolygonPoints(0, 0))
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.isStatic) return;
+
+    window.addEventListener('scroll', () => {
```
Comment (jamesgorrie) [inline common/views/components/WobblyEdge/WobblyEdge.js]: This should be debounced.

Comment (jamesgorrie) [reply common/views/components/WobblyEdge/WobblyEdge.js]: We should also kill the listener when we unmount:
https://stackoverflow.com/a/29726000

Comment (jamesgorrie) [reply common/views/components/WobblyEdge/WobblyEdge.js]: I wonder though if we could add it to the `onScroll` event of the component?

Comment (davidpmccormick) [reply common/views/components/WobblyEdge/WobblyEdge.js]: Good spot – in my haste to get it working I stripped out the debounce logic. I'll re-add it (and remove it on unmount).

Not sure I follow re. the `onScroll` even of the component. I think it's always going to have to be the `window` that we listen to.

File: common/views/components/WobblyEdge/WobblyEdge.js [hunk 3]
```diff
@@ -36,28 +37,35 @@ class WobblyEdge extends React.Component<Props, State> {
     };
   }
 
+  handleScroll = debounce(() => {
+    console.log('yoyoyoy');
```
Comment (jamesgorrie) [inline common/views/components/WobblyEdge/WobblyEdge.js]: Yo.

Comment (davidpmccormick) [reply common/views/components/WobblyEdge/WobblyEdge.js]: Yo.

**Pull request merged**